pycbc/io/hdf.py

```diff
@@ -168,13 +168,26 @@ def remove(self, idx):
             data[k] = np.delete(self.data[k], idx)
         return self._return(data=data)
 
+    def save(self, outname):
+        f = HFile(outname, "w")
+        for k in self.attrs:
+            f.attrs[k] = self.attrs[k]
+
+        for k in self.data:
+            f.create_dataset(k, data=self.data[k],
+                      compression='gzip',
+                      compression_opts=9,
+                      shuffle=True)
+        f.close()
+
 
 class StatmapData(DictArray):
-    def __init__(self, data=None, seg=None, attrs=None,
-                       files=None):
-        groups = ['stat', 'time1', 'time2', 'trigger_id1', 'trigger_id2',
-               'template_id', 'decimation_factor', 'timeslide_id']
-        super(StatmapData, self).__init__(data=data, files=files, groups=groups)
+    def __init__(self, data=None, seg=None, attrs=None, files=None,
+                 groups=['stat', 'time1', 'time2', 'trigger_id1',
```

Comment:
Codeclimate doesn't seem to like this default assignment.  Can we instead set groups to None as default and assign it to the right list of strings just before calling super.__init__ ?

Reply:
If `groups` never changes (is immutable), then the right solution might be to make it a tuple, not a list?

Reply:
`groups ` **here** doesn't change, but the `groups` it is set as in MultiifoStatmapData does; it is appended to on lines 228-229. The workaround for which would be a bit messy I think

Reply:
You could use tuples also in MultiifoStatmapData, I think.

Reply:
Although the default value of `groups` can be a tuple I think we should be able to override it by other (iterable) type .. no?

Reply:
If I change both then it works

Reply:
@GarethDaviesGW we usually use python lists for lists of named parameters so I'd prefer to stick with lists except where they are deprecated, ie here as a default kwarg value.

Reply:
Tom's suggestion implemented in b664cc0 - tested and works with tuple in kwargs and list overwriting this in MultiifoStatmapData